### PR TITLE
docs/guides/grub-migration: add grub migration guide

### PR DESCRIPTION
**Pull request by classabbyamp**
based on https://github.com/zbm-dev/zfsbootmenu/discussions/600#discussioncomment-8671690

### CODE REVIEW

File: docs/general/grub-migration.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    cd /zfsbootmenu/environments/bpool/ROOT/debian/mnt
+    mkdir -p /zfsbootmenu/environments/rpool/ROOT/debian/mnt/boot
+    cp -a . /zfsbootmenu/environments/rpool/ROOT/debian/mnt/boot
+
+6. Make sure the boot pool won't be mounted at next boot::
+
+    zfs set canmount=noauto bpool/ROOT/debian
+
+7. Exit the recover shell. You should now see your environment in the menu.
```
Comment (zdykstra) [inline docs/general/grub-migration.rst:44]: s/recover/recovery

**Review by zdykstra — APPROVED** [review]
It seems clear and concise, should be a good page to link for people.

**Comment by HankB** [comment]
`mount_zfs bpool/ROOT/debian`

I believe this should be `mount_zfs bpool/BOOT/debian`

Step numbering is off (or Step 3 is missing.)

Step 6 

```text
zfs set canmount=noauto bpool/ROOT/debian
```

should be

```text
zfs set canmount=noauto rpool/ROOT/debian
```

**Comment by classabbyamp** [comment]
> `mount_zfs bpool/ROOT/debian`
> 
> I believe this should be `mount_zfs bpool/BOOT/debian`

"Assuming your boot filesystem is `bpool/ROOT/debian`" 


> Step 6 
> 
> ```text
> zfs set canmount=noauto bpool/ROOT/debian
> ```
> 
> should be
> 
> ```text
> zfs set canmount=noauto rpool/ROOT/debian
> ```

no, it is disabling mounting of the bpool's /boot

**Comment by HankB** [comment]
Oof - I got the "fix" backwards. The step is

```text
zfs set canmount=noauto bpool/ROOT/debian
```

but that dataset does not exist.  I believe it should be 

```text
zfs set canmount=noauto bpool/BOOT/debian
```

`There is a `bpool/BOOT` dataset and `rpool/Root` but no `bool/ROOT`

**Comment by classabbyamp** [comment]
this is just a guide, you are expected to adapt it to fit your exact situation. we define what naming convention we assume throughout at the start so that readers can adapt it as needed.

**Pull request closed without merging**